### ISSUE TITLE
[css-typed-om][css-values] What context is CSSNumericValue.parse(...) considered to have for its parse?

### ISSUE THREAD

**Issue by weinig** [w3c/csswg-drafts#10753]
In most (I believe all other) cases, when parsing a `calc()` expression there is a known "context" which is needed to [determine the type](https://drafts.csswg.org/css-values-4/#determine-the-type-of-a-calculation) of percentage values. This is needed to figure out if the type should be a dimension with a percent hint (like when parsing a `width` property), or raw percentage (like when parsing `scale`).

With `CSSNumericValue.parse(...)` however, there is no specified context.

How are percentages supposed to be interpreted when parsed in this way?

**Comment by weinig** [comment]
cc @tabatkins, @fantasai

**Comment by tabatkins** [comment]
Correct, there isn't a specified context, so the paragraphs conditioned on the context don't apply.

*As far as I know*, the behavior should work out either way - if you have a value that works due to having a known context, then parsing it without a known context *then* matching it against a grammar should always work as well. Getting that right has been difficult and subtle; if you find an instance where it fails, it's a spec bug.
